### PR TITLE
Feat/Display location information publicly 

### LINKED ISSUE #2519: Location of in-person meeting not visible before booking
### Issue Summary

Location of in-person meeting not visible before booking

### Steps to Reproduce

1. Select a time for a meeting, e.g., at https://cal.com/marvin/coffee
2. Continue until you have the option to decide between "Zoom Video" and "In-person meeting"
3. Since you can't see where the in-person meeting is supposed to take place, no one will actually choose this option

→ It should be opt-out to display the actual location somewhere on the booking page, e.g., as "In-person meeting _in Amsterdam ℹ︎_" and by clicking on the location or ℹ︎, one can see the full address, so that people can see if this would actually be an option for them

### PR DESCRIPTION
**Pull request by sean-brydon**
## What does this PR do?

This gives the user the ability to toggle wether or not the location they have entered is displayed publicly. This PR also lets the user know where the location is before they book the meeting.

Fixes # (issue)
## Type of change

- [X] New feature (non-breaking change which adds functionality)

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/booking/pages/AvailabilityPage.tsx
Comment (sean-brydon) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:64]: Determine what to display to the user

File: apps/web/components/booking/pages/AvailabilityPage.tsx
Comment (sean-brydon) [inline apps/web/components/booking/pages/AvailabilityPage.tsx:356]: If the item in the array is the last item - don't display the "or"

File: apps/web/components/booking/pages/BookingPage.tsx
Comment (sean-brydon) [inline apps/web/components/booking/pages/BookingPage.tsx:194]: Removed as its been transferred into the Location types folder

File: apps/web/components/ui/form/CheckboxField.tsx
Comment (sean-brydon) [inline apps/web/components/ui/form/CheckboxField.tsx:11]: Update to the checkbox field 
-
As per recent designs checkboxes don't always have a "label" and are using the description as the label within the UI. Creates an easy an a11y way for this to be done in this component

Also added an Information Circle option - As this has also been present in a few designs

File: packages/app-store/locations.ts
Comment (sean-brydon) [inline packages/app-store/locations.ts:18]: Create this as a global type as it was being re-written across the application

File: apps/web/components/booking/pages/AvailabilityPage.tsx
Comment (PeerRich) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:64]: @zomars do we want to extract this eventually into app store? otherwise each video integration needs to be added here

Comment (PeerRich) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:64]: i think this can be solved without a switch

**Review by PeerRich — COMMENTED** [review]
good stuff

File: apps/web/components/booking/pages/AvailabilityPage.tsx
Comment (PeerRich) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:64]: opened a ticket for a future PR: https://github.com/calcom/cal.com/issues/2753

**Review by PeerRich — APPROVED** [review]
looks good! we can make the video providers dynamic later: https://github.com/calcom/cal.com/issues/2753

**Review by agustif — COMMENTED** [review]
Great work @sean-brydon !!
image

Looks great, only weird thing I had, was when I tried to edit the 30min event for pro user, before -updaging- the new locations. when I visited the preview, I was able to raise an error about locations being nullish?
image

Could you check that out? Only problem Is I wasn't able to reproduce again by removing locations and adding again so Idk...

I think a ? optional chaining operatior in `locations?.map` should fix this!

File: apps/web/pages/[user]/[type].tsx
Comment (agustif) [inline apps/web/pages/[user]/[type].tsx:267]: can we add a fallback here for when no locations set to map doesn't fail?
`locations: locations?.map((el) => {`
```suggestion
    locations: locations.map((el) => {
      // Filter out locations that are not to be displayed publicly
      const values = Object.values(AppStoreLocationType);
```

**Review by agustif — APPROVED** [review]
Awesome work on the feature Sean! Approved

This will make it much easier to make new apps that support the manual link process some how like around.co!

File: apps/web/pages/event-types/[type].tsx
Comment (emrysal) [inline apps/web/pages/event-types/[type].tsx:440]: This should be a Controller

```typescript
<Controller
  name="displayLocationPublicly"
  ....
  render={ ({ field: { onChange, value } }) => <CheckboxField ... onChange={(e) => onChange(e.target.checked)} />} />
```

File: apps/web/components/booking/pages/AvailabilityPage.tsx
Comment (emrysal) [reply apps/web/components/booking/pages/AvailabilityPage.tsx:65]: Can we use https://github.com/calcom/cal.com/blob/main/apps/web/lib/locationOptions.tsx here?

**Review by emrysal — APPROVED** [review]

File: apps/web/pages/team/[slug]/[type].tsx
Comment (zomars) [inline apps/web/pages/team/[slug]/[type].tsx:118]: Third time repeating this code feel like the need of abstraction. Or filtering way upstream.

**Review by zomars — COMMENTED** [review]
Nice work @sean-brydon

Some nits here and there but since you already have 3 approval this can be a follow up.

**Review by zomars — CHANGES_REQUESTED** [review]
For some reason this PR is breaking 9 tests locally. Will block merge to investigate.

**Comment by baileypumfleet** [comment]
Any update on this @zomars?

**Comment by zomars** [comment]
> Any update on this @zomars?

@sean-brydon will look into it.

File: apps/web/pages/[user]/[type].tsx
Comment (sean-brydon) [reply apps/web/pages/[user]/[type].tsx:265]: I've tried for the last two or so hours trying to get an elegant solution to this with prisma- prisma isn't the best with JSON filtering and it was getting quiet messy

File: apps/web/pages/d/[link]/[slug].tsx
Comment (sean-brydon) [reply apps/web/pages/d/[link]/[slug].tsx:148]: Abstracted away

**Review by zomars — APPROVED** [review]

**Pull request merged**
